### PR TITLE
Some fixes to python formatting

### PR DESCRIPTION
**Pull request by xylar**
In anticipation of supporting python 3, I was taking a look at what would need to be changed in a-prime.  I noticed some formatting issues and some incorrect use of `if __name__ == "__main__":...` directives that I thought needed to be cleaned up before we tackle python 3 support.

### CODE REVIEW

**Comment by xylar** [comment]
I also included a fix for #17 here because it kept annoying me every time I need to update...

**Comment by milenaveneziani** [comment]
Thanks @xylar for helping with this.
Yesterday, when I was debugging the problems @golaz was having with running some of the diagnostics, I realized that the permissions for some of the files (the bash scripts that launch the enso_atm_diags scripts) have been changed at some point by mistake.
This is fixed by doing 'chmod goa+x bash_scripts/*'.
Would you mind doing this within this PR? seems like a simple thing.

**Comment by kevans32** [comment]
Just ran this branch. Several plots still not showing up, we are looking into that.

**Comment by xylar** [comment]
@milenaveneziani, the fix you asked for is already the first commit in this PR.

**Comment by milenaveneziani** [comment]
ah, great. thanks.

**Comment by milenaveneziani** [comment]
I am testing this now with a different beta3 run, whose output seems to have the right permissions.

**Comment by xylar** [comment]
@milenaveneziani, if your test is currently running, you might want to start it over.  I messed up the acme-unified install by mistake and just restored it.  I wouldn't trust any errors you got in the last 30 minutes or so, since they could be my fault (just like @salilmahajan just reported).  Sorry!

**Comment by milenaveneziani** [comment]
All is working, I think I am getting everything right, judging from the plots that are being produced. So, I must have started the test before any updates in acme-unified. Thanks for the heads-up though.

**Comment by milenaveneziani** [comment]
As I told @salilmahajan by email, I am still not getting all of the Nino atm diagnostics, for reasons that I do not understand. But we should figure it out soon.

**Comment by xylar** [comment]
@milenaveneziani and @salilmahajan, if you're seeing the same missing plots with `master` as with this PR, I would suggest that we go ahead and merge these changes and seek a fix to the missing plots in a different PR.  If the missing plots are caused by this PR, that's another story.

**Comment by salilmahajan** [comment]
@xylar, @milenaveneziani, all the plots show up for the master for me for the beta3rc10 case for the atm and enso diags:

http://portal.nersc.gov/project/acme/salil/20171122.beta3rc10_1850.ne30_oECv3_ICG.edison_years1-10_vs_obs/

I will try it next using Xylar's branch.

**Comment by salilmahajan** [comment]
With Xylar's branch the feedbacks and lead-alf regression plots are missing: the same as Milena's test.

**Comment by milenaveneziani** [comment]
oh, thanks, I was starting to think it was just me..

As I mentioned in the email, the error I get is this:
```
Traceback (most recent call last):
  File "python/plot_regress_lead_lag_index_field.py", line 459, in <module>
    debug = debug)
  File "python/plot_regress_lead_lag_index_field.py", line 79, in plot_regress_lead_lag_index_field
    debug     = debug)
  File "/global/u2/m/milena/a-prime/python/get_regress_index_field.py", line 84, in get_regress_index_field
    field_no_ann = remove_seasonal_cycle_monthly_data(field, n_months_season, debug = debug)
UnboundLocalError: local variable 'field' referenced before assignment
```

File: python/get_regress_index_field.py [hunk 1]
```diff
@@ -21,49 +21,49 @@
...
+    if aggregate == 0 and no_ann == 1:
+        index_no_ann = remove_seasonal_cycle_monthly_data(index, n_months_season, debug = debug)
+        index = index_no_ann
+
+    if stdize == 1:
+        index_stddize = standardize_time_series(index)
+        index = index_stddize
 
 
         field, lat_reg, lon_reg, area_reg, units_out = read_monthly_data_ts(indir = indir[0],
```
Comment (xylar) [inline python/get_regress_index_field.py]: I believe this line is the problem.  The indentation is wrong and it ended up in the if statement above, which it should not have.

**Comment by xylar** [comment]
Okay, thanks to both of you for the clarification.  I believe I see the source of this problem (see above).

I believe have seen other `UnboundLocalError`s in other places.  Do you seen any with `master`?

**Comment by milenaveneziani** [comment]
@xylar: I will run with this fix and let you know. Thanks!

**Comment by xylar** [comment]
I am seeing the errors in #19 in both `master` and in this branch for the `20170915.beta2.A_WCYCL1850S.ne30_oECv3_ICG.edison.scratch` run.  Is this a beta2 issue?  Are you 2 seeing these as well?

**Comment by milenaveneziani** [comment]
We are running with beta3 (trying to also get new plots for Chris before the coupled group telecom).

Before, I used beta2 and I didn't have problems with master.

I started the new test with this updated branch: will let you know what I get.

**Comment by milenaveneziani** [comment]
@xylar, the test hasn't finished yet (it's doing the mpas stuff), but if I grep `Unbounded` into the new log files, this is what I see:
```
grep UnboundLocalError *.log
plot_enso_feedbacks_20171122.beta3rc10_1850.ne30_oECv3_ICG.edison_FLNS_vs_Nino3_ANN.log:UnboundLocalError: local variable 'field_name_ref' referenced before assignment
plot_enso_feedbacks_20171122.beta3rc10_1850.ne30_oECv3_ICG.edison_FSNS_vs_Nino3_ANN.log:UnboundLocalError: local variable 'field_name_ref' referenced before assignment
plot_enso_feedbacks_20171122.beta3rc10_1850.ne30_oECv3_ICG.edison_LHFLX_vs_Nino3_ANN.log:UnboundLocalError: local variable 'field_name_ref' referenced before assignment
plot_enso_feedbacks_20171122.beta3rc10_1850.ne30_oECv3_ICG.edison_RESSURF_vs_Nino3_ANN.log:UnboundLocalError: local variable 'field_list' referenced before assignment
plot_enso_feedbacks_20171122.beta3rc10_1850.ne30_oECv3_ICG.edison_SHFLX_vs_Nino3_ANN.log:UnboundLocalError: local variable 'field_name_ref' referenced before assignment
plot_enso_feedbacks_20171122.beta3rc10_1850.ne30_oECv3_ICG.edison_TS_vs_Nino3_ANN.log:UnboundLocalError: local variable 'field_name_ref' referenced before assignment
plot_meridional_avg_Tropical_Pacific_20171122.beta3rc10_1850.ne30_oECv3_ICG.edison_TS_MAM.log:UnboundLocalError: local variable 'var_expr' referenced before assignment
plot_meridional_avg_Tropical_Pacific_20171122.beta3rc10_1850.ne30_oECv3_ICG.edison_TS_SON.log:UnboundLocalError: local variable 'var_expr' referenced before assignment
plot_regr_20171122.beta3rc10_1850.ne30_oECv3_ICG.edison_RESSURF_vs_Nino3_ANN.log:UnboundLocalError: local variable 'field_list' referenced before assignment
plot_regr_20171122.beta3rc10_1850.ne30_oECv3_ICG.edison_RESSURF_vs_Nino3_DJF.log:UnboundLocalError: local variable 'field_list' referenced before assignment
plot_regr_20171122.beta3rc10_1850.ne30_oECv3_ICG.edison_RESSURF_vs_Nino3_JJA.log:UnboundLocalError: local variable 'field_list' referenced before assignment
plot_regr_20171122.beta3rc10_1850.ne30_oECv3_ICG.edison_RESSURF_vs_Nino3_MAM.log:UnboundLocalError: local variable 'field_list' referenced before assignment
plot_regr_20171122.beta3rc10_1850.ne30_oECv3_ICG.edison_RESSURF_vs_Nino3_SON.log:UnboundLocalError: local variable 'field_list' referenced before assignment
plot_regr_lead_lag_20171122.beta3rc10_1850.ne30_oECv3_ICG.edison_RESSURF_vs_Nino3.4_ANN.log:UnboundLocalError: local variable 'field_list' referenced before assignment
plot_stddev_Greater_Tropical_Pacific_20171122.beta3rc10_1850.ne30_oECv3_ICG.edison-COREv2_flux.RESSURF.ANN.log:UnboundLocalError: local variable 'field_list' referenced before assignment
plot_stddev_Greater_Tropical_Pacific_20171122.beta3rc10_1850.ne30_oECv3_ICG.edison-COREv2_flux.RESSURF.DJF.log:UnboundLocalError: local variable 'field_list' referenced before assignment
plot_stddev_Greater_Tropical_Pacific_20171122.beta3rc10_1850.ne30_oECv3_ICG.edison-COREv2_flux.RESSURF.JJA.log:UnboundLocalError: local variable 'field_list' referenced before assignment
plot_stddev_Greater_Tropical_Pacific_20171122.beta3rc10_1850.ne30_oECv3_ICG.edison-COREv2_flux.RESSURF.MAM.log:UnboundLocalError: local variable 'field_list' referenced before assignment
plot_stddev_Greater_Tropical_Pacific_20171122.beta3rc10_1850.ne30_oECv3_ICG.edison-COREv2_flux.RESSURF.SON.log:UnboundLocalError: local variable 'field_list' referenced before assignment
```

I will now try a simpler test (no ocn-ice) with master.

**Comment by milenaveneziani** [comment]
ok, here is what I found:
1) with this branch, some problems have been fixed with the most recent commit, but the Unbounded error shows up in a few more nino python scripts and so some other plots are affected. See details of the error in comment above.
2) with master, as you mentioned, we also get two Unbounded error:
```
/global/homes/m/milena/proj_acme/milena/coupled_diagnostics/20171122.beta3rc10_1850.ne30_oECv3_ICG.edison_vs_obs/20171122.beta3rc10_1850.ne30_oECv3_ICG.edison_years1-10_vs_obs.logs/plot_meridional_avg_Tropical_Pacific_20171122.beta3rc10_1850.ne30_oECv3_ICG.edison_TS_MAM.log:UnboundLocalError: local variable 'var_expr' referenced before assignment
/global/homes/m/milena/proj_acme/milena/coupled_diagnostics/20171122.beta3rc10_1850.ne30_oECv3_ICG.edison_vs_obs/20171122.beta3rc10_1850.ne30_oECv3_ICG.edison_years1-10_vs_obs.logs/plot_meridional_avg_Tropical_Pacific_20171122.beta3rc10_1850.ne30_oECv3_ICG.edison_TS_SON.log:UnboundLocalError: local variable 'var_expr' referenced before assignment
```
but these two are harmless, for some reason. All the plots show up as far as I can tell.

So, I would say, let us start taking care of 1), and then we can dig more to investigate 2).

**Comment by xylar** [comment]
@milenaveneziani, I think I've fixed the remaining 2 places where the tabbing was still messed up.  I'm testing again.

**Comment by kevans32** [comment]
ping us if you need more testers. At some point we need to update/run on rhea but this should not hold up any PRs

**Comment by milenaveneziani** [comment]
Thanks @xylar. I will give it another try myself.

**Comment by xylar** [comment]
My test passes, in that I only see the 2 error messages on the atmosphere side that @milenaveneziani sees (and that were also there on `master`).  @milenaveneziani, let me know if it goes well.

**Comment by milenaveneziani** [comment]
Test passed for me too.

**Comment by milenaveneziani** [comment]
Thanks so much @xylar!

**Pull request merged**